### PR TITLE
Implemented feature for group_graph hierarchical clustering functions that permits to specify the number of communities wanted

### PR DESCRIPTION
**Pull request by eferos93**
Added a new parameter `n_communities` for the functions
```
group_edge_betweenness
group_fast_greedy
group_leading_eigen
group_walktrap
```
that solves the issue #138 
Now it is then possible to control the number of communities outputted by the functions above.

### CODE REVIEW

File: R/group.R [hunk 1]
```diff
@@ -48,23 +48,32 @@ group_components <- function(type = 'weak') {
...
   weights <- enquo(weights)
   weights <- eval_tidy(weights, .E())
   # NULL in weights is for once respected despite a weight attribute
-  group <- as.integer(membership(cluster_edge_betweenness(graph = .G(), weights = weights, directed = directed)))
+  clusters <- cluster_edge_betweenness(graph = .G(), weights = weights, directed = directed)
+  if (!is.null(n_communities)) {
+    #clusters <- igraph::cut_at(clusters, no = n_communities)
+    return(clusters %>% igraph::cut_at(no = n_communities) %>%  as.integer %>% desc_enumeration)
+  }
+  group <- as.integer(membership(clusters))
```
Comment (thomasp85) [inline R/group.R:61]:
```suggestion
  if (!is.null(n_communities)) {
    group <- cut_at(clusters, no = n_communities)
  } else {
    group <- membership(clusters)
  }
  group <- as.integer(group)
```

Comment (thomasp85) [reply R/group.R:61]: I'd prefer something like the above for all of the implementations, along with a `@importFrom igraph cut_at` line in the docs in order to import the cut_at function in the namespace

**Review by thomasp85 — CHANGES_REQUESTED** [review]
This in general looks good, but I'd prefer a slightly different structure as described in the comment

**Comment by thomasp85** [comment]
also, I'd prefer `n_groups` over `n_communities` just to keep the naming internally consistent

File: R/group.R (hunk 1 above)
Comment (eferos93) [reply R/group.R:61]: Ok perfect, I will apply these changes and the documentation.

**Comment by eferos93** [comment]
@thomasp85 after compiling the documentation, the `man/group_graph.Rd` was updated as well as other .Rd files. Shall I commit them as well?

**Comment by thomasp85** [comment]
Sorry for slow response - just commit the files pertaining to your fix

**Comment by thomasp85** [comment]
Please add a news bullet and then this is ready for merging - thank you for your work on this

**Review by thomasp85 — APPROVED** [review]

**Comment by thomasp85** [comment]
Thanks!

**Pull request merged**